Checkpoints Collection TTL

kekachmar opened the pull request:
## Description

> This creates a Time-To-Live value for all documents stored in the checkpoints collection by adding a TTL index.

kekachmar committed:
test TTL creation

kekachmar committed:
make checkpoints collection TTL configurable

shubhi1092 committed:
Add restartOnCheckpointFailure flag for scribe, deli, scriptorium (#15482)

Add restartOnCheckpointFailure flag for scribe, deli, scriptorium

## Description

Services restart every time there is an exception in kafka checkpoint.
This PR allows to configure if it should restart or not on such
exceptions.

---------

Co-authored-by: Shubhangi Agarwal <shuagarwal@microsoft.com>

zhangxin511 commented on server/routerlicious/kubernetes/routerlicious/templates/fluid-configmap.yaml:
```diff
@@ -63,8 +63,9 @@ data:
-            "localCheckpointEnabled": {{ .Values.checkpoints.localCheckpointEnabled }}
-            },
+            "localCheckpointEnabled": {{ .Values.checkpoints.localCheckpointEnabled }},
+            "checkpointsTTL": 864000
```
Just nit, maybe do `checkpointsTTLInSeconds`?

zhangxin511 commented on server/routerlicious/packages/routerlicious-base/src/alfred/runnerFactory.ts:
```diff
@@ -227,6 +227,10 @@ export class AlfredResourcesFactory implements core.IResourcesFactory<AlfredReso
 			false,
 		);
+        
+        const checkpointsTTLSeconds = config.get("checkpoints:checkpointsTTL");
```
Nit, add a default or throw here for invalid/missing configs?

Josmithr committed:
build(devtools): Fix packageVersion.ts discrepency (#15517)

Version was not fully updated when package was added in #15490.

Updated via `npm run build` in the package directory.

CraigMacomber committed:
Update Readme (#15520)

## Description

Move general install and build instructions out of the Mac specific
section.

Remove use of lerna in descriptions of release groups as it is no longer
accurate.

Update list of release groups to account for changes with azure.

Josmithr committed:
refactor: Rename "client-debugger" directory to "devtools" (#15505)

Renames the directory to match updated nomenclature.

CraigMacomber committed:
Remove use of lerna in debug portmapping (#15514)

## Description

Replace use of lerna in debut portmapping tool with pnpm.

Replace use of child_process with spawnSync to avoid unneeded shell.

Add testing to and one test to package.

kekachmar committed:
set checkpoints ttl

ChumpChief committed:
Remove IHelpMessage, IQueueMessage, RemoteHelp and deprecate localHelp event (#15518)

CraigMacomber committed:
Update Dependency-Guidelines-Template (#15526)

## Description

Update Dependency-Guidelines-Template to explain how to use a "`>= <`
dependency range" since its not obvious what that means, and very
important people get it right.

CraigMacomber committed:
Update test-tools to 0.2.158186 (#15524)

## Description

Update test-tools to 0.2.158186.

This removes their use of lerna.

NicholasCouri committed:
Summarizer Election - Add logging for parent election as well as client election.  (#15522)

## Description

Add logging for parent change as well as client during client election
as well as when a new parent is elected. Based on the amount of logging
during the stress tests, we will probably need to consider throttling or
reducing the number of times we log it.

ChumpChief committed:
Remove bigBlobs secret feature from DataObject (#15527)

Josmithr committed:
test(devtools-view): Add SharedMatrix to test app (#15457)

Adds a usage of SharedMatrix by updating the emoji button view to be a
small 2x2 grid of emoji buttons.


![emoji-grid](https://user-images.githubusercontent.com/54606601/236335038-8f2e3c42-7e54-4bf5-8796-071c8f59422a.png)

CraigMacomber committed:
Update webpack (#15528)

## Description

Update webpack from 5.76.2 to 5.82.0

anthony-murphy committed:
MergeTree: Clean up handling for removed segments below the min seq (#15443)

# Description
Clean up and unify some handling of removed segments below the min seq.
Adds some testability improvement to visualize them.

related
[AB#1882](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/1882)

Co-authored-by: Tony Murphy <anthonm@microsoft.com>

anthony-murphy committed:
Prepare DDS for Protocol Defs 2.0 (#15383)

## Description
Add handling in the container loader package for typing improvements
made as part of https://github.com/microsoft/FluidFramework/pull/15353
which will come in protocol definitions 2.0.

---------

Co-authored-by: Tony Murphy <anthonm@microsoft.com>

ChumpChief committed:
Deprecate PureDataObject.getFluidObjectFromDirectory (#15530)

alexvy86 committed:
fix(devtools): Update @fluentui deps to get past the borked versions (#15515)

alexvy86 committed:
refactor(ci): Get npm feed URLs from centralized variable group (#15507)

tylerbutler committed:
build: Remove packageVersion files (#15512)

There are a lot of packages that generate a packageVersion.ts file but
that file is not used. I removed all the unused files and updated the
build scripts acordingly.

scottn12 committed:
[AzureClient] Increase container connect timeout in e2e tests (#15525)

## Description

Increase container connect timeout in azure e2e tests from 5s to 10s

noencke committed:
Make undoRedoManager optional in branches (#15519)

## Description

In additional to various cleanups around the branch and UndoRedoManager
APIs, this PR makes undo/redo "optional" for branches. A branch may be
constructed without an UndoRedoManager, in which case it does not do any
of the undo/redo related tracking. This will allow future branches that
don't need undo/redo to be constructed without the overhead.

jatgarg committed:
Only submit ops and signals when socket is connected (#15506)

## Description

Fix: https://dev.azure.com/fluidframework/internal/_workitems/edit/4174
Only submit ops and signals when socket is connected. This is to fix the
assert 0x244 which we were hitting in case connection is not yet
disposed but socket is already connected. This was happening in loop.
Scenario:
We have multiple containers in the given sessionId which share socket.
When socket disconnects all the containers need to go through
disconnection flow. During the disconnect flow of 1 container, when we
raise disconnect event on that container, the app tries to submit a
deletePresence signal on a different container which has not gone
through the disconnect flow yet and its connection is not yet disposed
while the socket is disconnected causing this assert and this exception
is recorded in the other container which emitted the "disconnect" event
which should be expected.

We should have dropped the signal in the happy path.

kekachmar committed:
Merge branch 'checkpointsTTL' into checkpointTTL

tylerbutler reviewed (changes requested):
This PR seems to have a lot of unrelated changes, especially outside of the server folder. Can you please rebase on the latest main before continuing?
